### PR TITLE
Adds googletest builder

### PR DESCRIPTION
**Pull request by KMarkert**
This PR adds [googletest](https://github.com/google/googletest) framework.

### CODE REVIEW

File: G/googletest/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    LibraryProduct("libgtest_main", :libgtest_main),
+    LibraryProduct("libgtest", :libgtest)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline G/googletest/build_tarballs.jl:39]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)

```

File: G/googletest/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+mkdir build && cd build
+cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release .. -DBUILD_GMOCK=OFF
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+platforms = expand_cxxstring_abis(platforms)
```
Comment (giordano) [inline G/googletest/build_tarballs.jl:26]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: G/googletest/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/google/googletest.git", "703bd9caab50b139428cea1aaff9974ebee5742e")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd googletest/
```
Comment (giordano) [inline G/googletest/build_tarballs.jl:16]:
```suggestion
cd $WORKSPACE/srcdir/googletest
```

**Comment by giordano** [comment]
Not needed for #2643, but it doesn't hurt to have it either.  Thanks!

**Comment by KMarkert** [comment]
Thank you for your help!

**Pull request merged**